### ISSUE TITLE
[BUG] RKE2 cluster provisioning faild in the embedded Rancher (only master-head)

### ISSUE THREAD

**Issue by futuretea** [harvester/harvester#3973]
**Describe the bug**

![image](https://github.com/harvester/harvester/assets/15064560/17f2867a-fca1-4c86-ba2d-471b673a3f66)

**To Reproduce**
1.  Install a Harvester cluster (master-head)
2.  change Harvester setting `rancher-manager-support` to true
3.  create rke2 cluster in the embedded Rancher's `Cluster Management`

**Expected behavior**

RKE2 cluster provisioning successfully

**Support bundle**

**Environment**
 - Harvester ISO version: master-2d5a3a9-head
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):

**Additional context**

when creating an rke2 cluster, Rancher creates a job, job will create a rancher-machine pod, the pod will download the harvester node driver from Rancher. This error message `--driver-download-url=/assets/docker-machine-driver-harvester ` tells us that the pod is downloading from a wrong address `/assets/docker-machine-driver-harvester`, the correct address should be `https://10.11.6.10/assets/docker-machine-driver-harvester`

The missing part is the value of Rancher's setting `server-url` and it should be `https://10.11.6.10`

but check the value of Rancher's setting `server-url`, it's empty

Rancher's server-url is configured from the frontend through the API when the user first login.

In the master version, the frontend does not send the server-url setting updating request

**Comment by guangbochen** [comment]
We must set/allow to set the `server-url` since this is needed for the multi-cluster feature, will need to check if we can sync it via the backend automatically using the VIP address.

**Comment by n313893254** [comment]
Hi, @ibrokethecloud. Do we need any modifications on the UI part?

**Comment by TachunLin** [comment]
Verified **fixed** on `v1.2-27934bc3-head` (06/06). Close this issue.

### Result
* Checked the `server-url` match the Harvester VIP 
```
harv12bp-0606:~ # kubectl get setting.management server-url
NAME         VALUE
server-url   https://192.168.122.111
```

* Can provision RKE2 cluster on the embedded Rancher in Harvester v1.2-head (v1.25.9+rke2r1)
![image](https://github.com/harvester/harvester/assets/29251855/6cfde7de-95f1-4404-9376-ef8e30da78d2)

### Test Information
* Test Environment: Single nodes Harvester on local kvm machine
* Harvester version: `v1.2-27934bc3-head` (06/06)

### Verify Steps
1. Prepare image and create vlan1 network on Harvester
2. Enable the `rancher-manager-support` setting in Harvester
3. Click the hamburger menu on the left upper side 
4. Enter Cluster Management 
5. Create a cloud credential and select `local` Harvester
6. Create a RKE2 cluster 
7. Provide the login credential in user data 
  ```
  password: 123456
  chpasswd: { expire: False }
  ssh_pwauth: True
  ```
8. Check can provision the RKE2 cluster successfully
